### PR TITLE
[DNM]Adds YouTube comments[GONE WRONG][GONE SEXUAL] 

### PR DESCRIPTION
**Pull request by SamHPurp**
You may now comment on articles on news feeds. Hopefully providing a nice RP buff for Librarians/Reporters ~~/Clowns~~

This also allows for the scribbles on the back of photos to act as a 'Caption' when submitted for a news article.

Furthermore this adds admin interactions to they can delete comments if necessary.

I was unable to test this due to no local SQL running... so a good review appreciated. ~~it compiles ship it~~

:cl: Purpose2
add: Adds a system for leaving comments on Newscaster Articles. Go get that big story, and be rubbished as hack by the crew!
add: Writing on the backs of photos adds this as a 'Caption' for news article photos.
/ :cl:

### CODE REVIEW

**Comment by FlattestGuitar** [comment]
First

**Comment by imsxz** [comment]
like favorite and subscribe for more content

File: code/__HELPERS/logging.dm [hunk 1]
```diff
@@ -82,6 +82,10 @@
 	if (config.log_pda)
 		diary << "\[[time_stamp()]]CHAT: [text]"
 
+/proc/log_comment(text)
+	if (config.log_pda) //reusing as not worth its own config option really
```
Comment (tigercat2000) [inline code/__HELPERS/logging.dm]: No spacing between control statements and their brackets please. I do see that the above is also incorrect, and it is wrong.

Comment (SamHPurp) [inline code/__HELPERS/logging.dm]: - [ ] Keep as is?

- [ ] Update to correct standard?

- [ ] Update to correct standard across the board?

**Comment by granodd** [comment]
Finally I can be a famous YouTube streamer in Space Station.

File: code/game/machinery/newscaster.dm [hunk 2]
```diff
@@ -555,6 +567,18 @@ var/list/obj/machinery/newscaster/allCasters = list() //Global list that will co
 			viewing_channel = FC
 			screen = NEWSCASTER_CENSOR_FC
 
+	else if(href_list["new_comment"])
+		var/datum/feed_message/FM = locate(href_list["new_comment"])
+		var/cominput = copytext(stripped_input(usr, "Write your message:", "new comment", null),1,141)
+		if(cominput)
```
Comment (KasparoVy) [inline code/game/machinery/newscaster.dm]: Used correct standard everywhere else but that one line

**Comment by Citinited** [comment]
Excellent PR, I went ahead and hit the little bell to get notified of new videos 👍

**Comment by alexkar598** [comment]
did a tag got created just for this pr???

**Comment by Anasari** [comment]
insert randomly irrelevant political comment about the us and communism here

**Comment by outbools** [comment]
Space communism is cool

File: code/modules/admin/topic.dm [hunk 3]
```diff
@@ -3240,6 +3240,18 @@
...
+	else if(href_list["ac_del_comment"])
+		var/datum/feed_comment/FC = locate(href_list["ac_del_comment"])
+		var/datum/feed_message/FM = locate(href_list["ac_del_comment_msg"])
+		FM.comments -= FC
+		qdel(FC)
+		src.access_news_network()
+
+	else if(href_list["ac_lock_comment"])
+		var/datum/feed_message/FM = locate(href_list["ac_lock_comment"])
+		FM.locked ^= 1
```
Comment (tigercat2000) [inline code/modules/admin/topic.dm]: This should just be `FM.locked = !FM.locked` for ease of readability.

**Comment by tigercat2000** [comment]
Stale, and pull request currently does not function. Please update this by 2017-10-1 or it will be closed; You are free to resubmit it at a later date should it take you longer than that to continue development.

**Comment by taukausanake** [comment]
@SamHPurp What happened?

**Pull request closed without merging**